ext/pdo/pdo_stmt.c

```diff
@@ -772,88 +746,78 @@ static bool do_fetch(pdo_stmt_t *stmt, zval *return_value, enum pdo_fetch_type h
 			break;
 
 		case PDO_FETCH_OBJ:
-			object_init_ex(return_value, ZEND_STANDARD_CLASS_DEF_PTR);
+			ce = zend_standard_class_def;
+			object_init_ex(return_value, zend_standard_class_def);
 			break;
 
 		case PDO_FETCH_CLASS:
+			ce = stmt->fetch.cls.ce;
 			if (flags & PDO_FETCH_CLASSTYPE) {
-				zval val;
-				zend_class_entry *cep;
-
-				old_ce = stmt->fetch.cls.ce;
-				ZVAL_COPY_VALUE(&old_ctor_args, &stmt->fetch.cls.ctor_args);
-				old_arg_count = stmt->fetch.cls.fci.param_count;
-				do_fetch_opt_finish(stmt, 0);
-
-				fetch_value(stmt, &val, column_index_to_fetch++, NULL);
-				if (Z_TYPE(val) != IS_NULL) {
-					if (!try_convert_to_string(&val)) {
-						return 0;
-					}
-					if ((cep = zend_lookup_class(Z_STR(val))) == NULL) {
-						stmt->fetch.cls.ce = ZEND_STANDARD_CLASS_DEF_PTR;
-					} else {
-						stmt->fetch.cls.ce = cep;
+				zval ce_name_from_column;
+				fetch_value(stmt, &ce_name_from_column, column_index_to_fetch++, NULL);
+				if (Z_TYPE(ce_name_from_column) == IS_STRING) {
```

Comment:
If the type is not string, then we silently ignore it? That seems not nice... should warn or throw or something

Reply:
Annoyingly, due to the previous usage of `try_convert_to_string()` it would accept integers/null/floats convert them to string, and because no class could possibly exist set the CE to stdClass.

Reply:
Could we defer this to a follow-up PR? I would like to keep the test changes to a minimum, and I wonder if this problem might be caught earlier (e.g. if we already have some column metadata).

Reply:
Yes, let's defer